rts/lra/lra-test/basic/src/test/java/io/narayana/lra/arquillian/resource/TestLRAParticipant.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2019, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package io.narayana.lra.arquillian.resource;
+
+import org.eclipse.microprofile.lra.annotation.Compensate;
+import org.eclipse.microprofile.lra.annotation.ws.rs.LRA;
+
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import java.net.URI;
+
+@Path("/participant")
+public class TestLRAParticipant {
+    @PUT
+    @Path("/compensate")
+    @Compensate
+    public Response compensate(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+        return Response.status(Response.Status.ACCEPTED).build();
+    }
+
+    @PUT
+    @Path("/complete")
+    @Compensate
+    public Response complete(@HeaderParam(LRA.LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+        return Response.status(Response.Status.ACCEPTED).build();
```

Comment:
since there is no `@Status` method, `@Complete` will be called repeatedly which means that the LRA started in testFinishLRA is never finished. This will "leak" unfinished LRA to other tests which may propose problems in the future. However, we can adjust this in a similar way as in the TCK when we introduce a custom TCK interface for retriggering recovery.

Reply:
The test and coordinator will be torn down so it's not needed

Reply:
But you comment is something to look out for if we end up adding other Narayana specific tests

Reply:
exactly, for now, it's ok

Reply:
+1, I would favour only the test code that is strictly necessary to test the use case be added for the aid of reviewing.